src/modules/products/routes/products.ts

```diff
@@ -1,9 +1,23 @@
-import { Router } from "express";
+import express from 'express';
+import * as ProductController from '../controller/productController';
+import imageUpload  from "../../../middleware/image-upload/multer";
 
-const router = Router();
 
-router.get("/api/products", (request, response) => {
-    return response.status(200).send('products router');
-});
+const router = express.Router();
+
+// POST /products - Create a new product
+router.post('/products', imageUpload, ProductController.createProduct);
+
+// GET /products/:id - Retrieve a product by its ID
+router.get('/products/:id', ProductController.getProductById);
+
+// GET /products - Retrieve all products
+router.get('/products', ProductController.getAllProducts);
+
+// PUT /products/:id - Update a product by its ID
+router.put('/product/:id', ProductController.updateProductById);
+
+// DELETE /products/:id - Delete a product by its ID
+router.delete('/product/:id', ProductController.deleteProductById);
```

Comment:
We have 
GET /products/:id 
but 
DELETE /product/:id. 
I would suggest using either plural for all the endpoints, or use the same formatting for getting/putting/deleting by id. Otherwise it can be confusing.

Reply:
let's use plural for all endpoints